### PR TITLE
Added enum to string conversion, tests and sample usage.

### PR DESCRIPTION
**Pull request by linus-sherrill**

### CODE REVIEW

File: vital/config/tests/test_config_block.cxx [hunk 1]
```diff
@@ -901,3 +903,34 @@ IMPLEMENT_TEST(set_value_description)
...
+  using namespace kwiver::vital;
+
+  config_block_sptr const config = config_block::empty_config();
+
+  config->set_value("keya", "three");
+  config->set_value("keyb", "foo");
+
+  int val = config->get_enum_value < my_ec >( "keya" );
+  TEST_EQUAL("enum value", val, 3 );
+
```
Comment (collinsr) [inline vital/config/tests/test_config_block.cxx]: What would happen if you tried config->get_enum_value< my_ec >( "non-existent-key" ) ?

File: vital/config/tests/test_config_block.cxx [hunk 2]
```diff
@@ -901,3 +903,34 @@ IMPLEMENT_TEST(set_value_description)
...
+  EXPECT_EXCEPTION( std::runtime_error,
+                    config->get_enum_value < my_ec >( "keyb" ),
+                    "enum conversion error");
+
```
Comment (collinsr) [inline vital/config/tests/test_config_block.cxx]: Maybe add another expect_exception for trying to retrieve a non-existent key (e.g. get_enum_value<my_ec>("foo"))

Comment (linus-sherrill) [inline vital/config/tests/test_config_block.cxx]: Thanks for the review. Added an additional test.

**Comment by hughed2** [comment]
Can one of the admins verify this patch?

**Comment by hughed2** [comment]
Can one of the admins verify this patch?

**Comment by hughed2** [comment]
Can one of the admins verify this patch?

**Comment by Purg** [comment]
Please cleanup branch on github remote.

**Pull request merged**